CONTRIBUTING.rst

```diff
@@ -0,0 +1,160 @@
+
+* Use a branching workflow similar to the one described at
+  http://progit.org/book/ch3-4.html.
+
+* Keep your own "master" and "develop" branches in sync with the blessed
+  repository's "master" and "develop" branches. Specifically, do not push your
+  own commits directly to your "master" and "develop" branches.
+
+* Any commit should *pass all tests* (see `Running Tests`_).
+
+* See the `An Example`_ section below for a full walk through
+
+Issuing a Pull Request
+======================
+
+* When you are ready to move changes from one of your topic branches into the
+  "develop" branch, it must be reviewed and accepted by another developer.
+
+* You may want to review this `tutorial
+  <https://help.github.com/articles/using-pull-requests/>`_ before you make a
+  pull request to the develop branch.
+
+Reviewing a Pull Request
+========================
+
+* Look over the code.
+
+  * Check that it meets `our style guidelines
+    <http://fuelcycle.org/kernel/pr_review.html>`_.
+
+  * Make inline review comments concerning improvements.
+
+* Wait for the Continuous Integration service to show full test passage
+
+* Click the green "Merge Pull Request" button
+
+  * Note: if the button is not available, the requester needs to merge or rebase
+    from the current HEAD of the blessed's "develop" (or "master") branch
+
+Cautions
+========
+
+* **NEVER** merge the "master" branch into the "develop" branch. Changes should
+  only flow *to* the "master" branch *from* the "develop" branch.
+
+* **DO NOT** rebase any commits that have been pulled/pushed anywhere else other
+  than your own fork (especially if those commits have been integrated into the
+  blessed repository.  You should NEVER rebase commits that are a part of the
+  'master' branch. *If you do, we will never, ever accept your pull request*.
+
+An Example
+==========
+
+Introduction
+------------
+
+As this type of workflow can be complicated to converts from SVN and very complicated
+for brand new programmers, an example is provided.
+
+For the sake of simplicity, let us assume that we want a single "sandbox" branch
+in which we would like to work, i.e. where we can store all of our work that may not
+yet pass tests or even compile, but where we also want to save our progress. Let us
+call this branch "Work". So, when all is said and done, in our fork there will be
+three branches: "Master", "Develop", and "Work".
+
+Acquiring Cyclus and Workflow
+-----------------------------
+
+We begin with a fork of the main ("blessed") Cyclus repository. After initially forking
+the repo, we will have two branches in our fork: "Master" and "Develop".
+
+Acquiring a Fork of the Cyclus Repository
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A fork is *your* copy of Cyclus. Github offers an excellent
+`tutorial <http://help.github.com/fork-a-repo/>`_ on how to set one up. The rest of this
+example assumes you have set up the "upstream" repository as cyclus/core. Note that git
```

Comment:
(Maybe not super important but I would say we need to distinguish local and remote before the commands below)